storage/libradb/src/schema/jellyfish_merkle_node/mod.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines physical storage schema for nodes in state Jellyfish Merkle tree.
+//! Node is identified by [NodeKey](JellyfishMerkle::node_type::NodeKey).
+//! ```text
+//! |<----key--->|<-----value----->|
+//! |  node_key  | serialized_node |
+//! ```
+
+use crate::schema::{ensure_slice_len_gt, JELLYFISH_MERKLE_NODE_CF_NAME};
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+use failure::prelude::*;
+use jellyfish_merkle::node_type::{Node, NodeKey};
+use schemadb::{
+    define_schema,
+    schema::{KeyCodec, ValueCodec},
+};
+use sparse_merkle::nibble_path::NibblePath;
+use std::{io::Write, mem::size_of};
+use types::transaction::Version;
+
+define_schema!(
+    JellyfishMerkleNodeSchema,
+    NodeKey,
+    Node,
+    JELLYFISH_MERKLE_NODE_CF_NAME
+);
+
+impl KeyCodec<JellyfishMerkleNodeSchema> for NodeKey {
+    fn encode_key(&self) -> Result<Vec<u8>> {
+        let mut encoded_key = vec![];
+        encoded_key.write_u64::<BigEndian>(self.version())?;
+        encoded_key.write_u8(self.nibble_path().num_nibbles() as u8)?;
```

Comment:
Have you thought of storing `is_odd` instead of this so that it's (a tiny little) more compressible physically and the decode code gets simpler. I do see this is more open to adding new fields to the key.. not sure if that is your concern.